hydra-cluster/src/Hydra/Cluster/Scenarios.hs

```diff
@@ -118,10 +118,20 @@ singlePartyHeadFullLifeCycle tracer workDir node@RunningNode{networkId} hydraScr
         guard $ v ^? key "tag" == Just "HeadIsClosed"
         guard $ v ^? key "headId" == Just (toJSON headId)
         v ^? key "contestationDeadline" . _JSON
+      time <- getCurrentTime
       -- Expect to see ReadyToFanout within 600 seconds after deadline.
       -- XXX: We still would like to have a network-specific time here
-      remainingTime <- diffUTCTime deadline <$> getCurrentTime
-      waitFor tracer (truncate $ remainingTime + 60) [n1] $
```

Comment:
The problem is the conversion of a `NominalDiffTime -> Natural` done by `truncate`. A better solution to this problem would be to change the type of `waitFor` (and all its related functions) to take a `NominalDiffTime`. The ultimate usage site for this time is when calling `timeout` which does not even take a `Natural` but `Int` (of micro seconds). So there should never be an underflow problem. In this solution we need to check what `timeout` does with a negative number, but likely it will just produce `Nothing` and we need to treat this differently specially.

Reply:
Yes that would be better. If `timeout` receives a negative number it will basically just run the inner computation immediately and wrap it in `Just` 

```
timeout :: Int -> IO a -> IO (Maybe a)
timeout n f
    | n <  0    = fmap Just f
    ...
```